subsys/bluetooth/host/id.c

```diff
@@ -1785,7 +1785,15 @@ int bt_id_set_scan_own_addr(bool active_scan, uint8_t *own_addr_type)
 	}
 
 	if (IS_ENABLED(CONFIG_BT_PRIVACY)) {
+
+		if (BT_FEAT_LE_PRIVACY(bt_dev.le.features)) {
+			*own_addr_type = BT_HCI_OWN_ADDR_RPA_OR_RANDOM;
+		} else {
+			*own_addr_type = BT_ADDR_LE_RANDOM;
+		}
```

Comment:
Since this code is just moved, I won't block this PR on this, but something is out of place here. 

Why is `own_addr_type` being set to one of 2 values, from each their own value spaces? The `BT_HCI_*` address types are not comparable to the `BT_ADDR_*` types:

```c
#define BT_ADDR_LE_PUBLIC       0x00
#define BT_ADDR_LE_RANDOM       0x01
#define BT_ADDR_LE_PUBLIC_ID    0x02
#define BT_ADDR_LE_RANDOM_ID    0x03
#define BT_ADDR_LE_UNRESOLVED   0xFE
#define BT_ADDR_LE_ANONYMOUS    0xFF
```
```c
#define BT_HCI_OWN_ADDR_RPA_OR_PUBLIC   0x02
#define BT_HCI_OWN_ADDR_RPA_OR_RANDOM   0x03
#define BT_HCI_OWN_ADDR_RPA_MASK        0x02
```


I'll create a follow-up PR that defines the missing HCI types and use those instead
![image](https://github.com/user-attachments/assets/0e3b8381-3c7c-46d8-90db-e52f3b39e1b6)

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/79645